src/link/sdas_obj.cpp

```diff
@@ -826,7 +826,37 @@ void sdobj_ReadFile(FileStackNode const &where, FILE *file, std::vector<Symbol>
 		std::unique_ptr<Section> &section = entry.section;
 
 		// RAM sections can have a size, but don't get any data (they shouldn't have any)
-		if (entry.writeIndex != section->size && entry.writeIndex != 0)
+		if (section->type != SECTTYPE_INVALID) {
+			auto const &typeInfo = sectionTypeInfo[section->type];
+			// Otherwise, how would the type already be known at this point?
+			assume(section->isAddressFixed);
+
+			if (!sect_HasData(section->type)) {
+				if (!section->data.empty()) {
+					fatal(
+					    &where,
+					    lineNo,
+					    "\"%s\" is implicitly defined as a %s section (being at address $%04" PRIx16
+					    "), but it has data! (Was a bad `__at()` value used?)",
```

Comment:
Is "`__at()`" something that GBDK/SDCC users would know the meaning of? Something they write themselves?

Reply:
Yeah, it's a special C attribute: `static char __at(0xC0DE) var;`.